src/databricks/labs/ucx/source_code/base.py

```diff
@@ -1,15 +1,20 @@
 from __future__ import annotations
 
+import sys
+
 from abc import abstractmethod
 from collections.abc import Iterable
 from dataclasses import dataclass
 from pathlib import Path
 
 from astroid import NodeNG  # type: ignore
-from typing_extensions import Self
 
 from databricks.sdk.service import compute
 
+if sys.version_info < (3, 11):
+    from typing_extensions import Self
```

Comment:
@asnare what do you think ?

Reply:
The answer is that of-course that we can: it's mainly a correctness thing for the return types of factory methods so that the type checking knows the concrete class being returned (`Failure`, `Warning`, etc) rather than only the base type (`Advice`). Mypy copes with this reasonably well although PyCharm/IntelliJ gets a bit confused and falsely highlights some code even though it's fine.

I normally lean on the side of correctness because I want the tooling to help me as much as possible, but I also understand the plumbing here is a bit smelly (even if it's idiomatic).

Reply:
let's get rid of `typing_extensions` from our code. the less external libraries we depend on, the better.